src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> node = getNode(index);
+        Node<T> newNode = new Node<>(node.prev, value, node);
+        if (index == 0) {
+            first = newNode;
+        } else {
+            node.prev.next = newNode;
+        }
+        node.prev = newNode;
+        size++;
```

Comment:
it is better to implement this part without else
```suggestion
        if (index == 0) {
            first = newNode;
            size++;
            return;
        }
        node.prev.next = newNode;
        node.prev = newNode;
        size++;
```